### PR TITLE
cmake: 3.27.8 -> 3.28.1

### PR DESCRIPTION
**Pull request by gracicot**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Reopened version of #275075

Previous people that participated in previous PR: @AndersonTorres @2xsaiko 

Sorry for the noise caused by the previous PR

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by 2xsaiko** [comment]
I should have maybe stressed more that you should follow the steps in that link. Sorry! 😅

**Comment by gracicot** [comment]
I thought I did, maybe I did something wrong by mistake 😭

It's okay, I'll be more careful next time, sorry again for the trouble

File: pkgs/by-name/cm/cmake/package.nix [hunk 1]
```diff
@@ -96,7 +96,8 @@ stdenv.mkDerivation (finalAttrs: {
   ++ lib.optional useOpenSSL openssl
   ++ lib.optional cursesUI ncurses
   ++ lib.optional qt5UI qtbase
-  ++ lib.optional (stdenv.isDarwin && !isMinimalBuild) SystemConfiguration;
+  ++ lib.optional (stdenv.isDarwin && !isMinimalBuild) SystemConfiguration
+  ++ lib.optional (stdenv.isDarwin) CoreServices;
```
Comment (AndersonTorres) [inline pkgs/by-name/cm/cmake/package.nix:100]:
```suggestion
  ++ lib.optional (stdenv.isDarwin) CoreServices
  ++ lib.optional (stdenv.isDarwin && !isMinimalBuild) SystemConfiguration;
```

A bit smaller diff

File: pkgs/by-name/cm/cmake/005-remove-systemconfiguration-dep.diff [hunk 2]
```diff
@@ -1,10 +1,10 @@
 diff -Naur cmake-3.27.7/Utilities/cmcurl/CMakeLists.txt cmake-3.27.7-new/Utilities/cmcurl/CMakeLists.txt
 --- cmake-3.27.7/Utilities/cmcurl/CMakeLists.txt	2023-10-06 10:08:35.000000000 -0300
```
Comment (gracicot) [inline pkgs/by-name/cm/cmake/005-remove-systemconfiguration-dep.diff:2]: I changed the diffs manually, so it means those headers didn't change. Should I regenerate from scratch or hand written diffs are fine?

Comment (AndersonTorres) [reply pkgs/by-name/cm/cmake/005-remove-systemconfiguration-dep.diff:2]: I personally prefer them generated via `diff old-dir new-dir`.

Comment (gracicot) [reply pkgs/by-name/cm/cmake/005-remove-systemconfiguration-dep.diff]: Done! I actually edited the CMake source and ran `diff -Naur cmake-3.28.1/... cmake-3.28.1-new/...` again to get this output

**Comment by AndersonTorres** [comment]
@gracicot please rebase

**Comment by gracicot** [comment]
It should be good to go! Thank you for the review.

If it would happen that this particular CMake version create problems, there is 3.28.2 coming out in a few days. It has merged some regression fixes.

**Comment by gracicot** [comment]
@AndersonTorres ofborg seem to fail since I rebase. I'm not sure if it's caused by this PR or not, as it was not causing errors before the rebase.

**Comment by AndersonTorres** [comment]
It was the same exact error on https://github.com/NixOS/nixpkgs/pull/285472 and https://github.com/NixOS/nixpkgs/pull/286473.

Three strikes? It looks like a bug on Staging!

**Comment by gracicot** [comment]
Let me know if there's anything I can do to help getting this PR merged, other than keeping it rebased 😄

**Comment by AndersonTorres** [comment]
> Let me know if there's anything I can do to help getting this PR merged, other than keeping it rebased 😄

Seeking mergers on Discourse, Discord, Matrix, Telegram, e-mail, telephone, snail mail or in person is a good idea.

**Pull request merged**